File: Assignment9/src/Repo/WorkspaceRepo.js
const load = async auth => {
  const url = 'https://cse118.com/api/v2/workspace';
  const headers = {
    Accept: 'application/json',
    'Content-Type': 'application/json',
    Authorization: `Bearer ${auth}`,
  };

  const response = await fetch(url, {
    method: 'GET',
    headers: headers,
  });

  const json = await response.json();
  return json;
};

export {load};

const add = async (auth, name) => {
  console.log('adding workspace');
  const url = 'https://cse118.com/api/v2/workspace';
  const headers = {
    Accept: 'application/json',
    'Content-Type': 'application/json',
    Authorization: `Bearer ${auth}`,
  };

  const data = {name};
  const jsonData = JSON.stringify(data);

  const response = await fetch(url, {
    method: 'POST',
    headers: headers,
    body: jsonData,
  });

  const responseData = await response.json();
  return responseData;
};

export {add};

const reset = async auth => {
  const url = 'https://cse118.com/api/v2/reset';
  const headers = {
    Accept: 'application/json',
    'Content-Type': 'application/json',
    Authorization: `Bearer ${auth}`,
  };

  const response = await fetch(url, {
    method: 'PUT',
    headers: headers,
  });

  const responseData = await response;
  console.log('Response');
  console.log(response);
  return responseData;
};

export {reset};

const Delete = async (auth, id) => {
  console.log('deleting workspace');
  console.log('id');
  const url = `https://cse118.com/api/v2/workspace/${id}`;
  const headers = {
    Accept: 'application/json',
    'Content-Type': 'application/json',
    Authorization: `Bearer ${auth}`,
  };

  const response = await fetch(url, {
    method: 'DELETE',
    headers: headers,
  });

  // The Swift code does not use the response data, so we can omit the next two lines in JavaScript
  console.log('Here');
  const responseData = await response;
  console.log(responseData);
  return responseData;
};

export {Delete};

File: Assignment6-1.Submission/Assignment6/Repo/WorkspaceRepo.swift
//
//  WorkspaceRepo.swift
//  Assignment6
//
//  Created by Abhinav Sunil on 5/16/23.
//

import Foundation


struct WorkspaceRepo {
    static func load(auth: String) async throws -> [Workspace] {
       // sleep(1) 
        let url = URL(string: "https://cse118.com/api/v2/workspace")!
        var request = URLRequest(url: url)
        request.setValue("application/json", forHTTPHeaderField: "Accept") // Set the Accept header
        request.setValue("Bearer \(auth)", forHTTPHeaderField: "Authorization") // Set the Authorization header

        let (data, _) = try await URLSession.shared.data(for: request) // Use the prepared request
        
        
        return try JSONDecoder().decode([Workspace].self, from: data)
    }
    
    static func add(auth: String, name: String) async throws {
        let url = URL(string: "https://cse118.com/api/v2/workspace")!
        var request = URLRequest(url: url)
        let Name = ["name": name]
        let jsonData = try JSONEncoder().encode(Name)
        
        
        request.setValue("application/json", forHTTPHeaderField: "Accept") // Set the Accept header
        request.setValue("Bearer \(auth)", forHTTPHeaderField: "Authorization")
        request.httpMethod = "POST"
        request.httpBody = jsonData
        request.setValue("application/json", forHTTPHeaderField: "Content-Type")
        
        
        
        
        let (data, _) = try await URLSession.shared.data(for: request) // Use the prepared request
        
//        if let jsonString = String(data: data, encoding: .utf8) {
//            print(jsonString)
//        }
        
    }
    
    static func delete(auth: String, id: String) async throws {
        let url = URL(string: "https://cse118.com/api/v2/workspace/\(id)")!
        var request = URLRequest(url: url)
        
        
        request.setValue("application/json", forHTTPHeaderField: "Accept") // Set the Accept header
        request.setValue("Bearer \(auth)", forHTTPHeaderField: "Authorization")
        request.httpMethod = "DELETE"
        
        let (data, _) = try await URLSession.shared.data(for: request)
        
    }
    
    
    static func reset(auth: String) async throws {
        let url = URL(string: "https://cse118.com/api/v2/reset")!
        var request = URLRequest(url: url)
        
        
        request.setValue("application/json", forHTTPHeaderField: "Accept") // Set the Accept header
        request.setValue("Bearer \(auth)", forHTTPHeaderField: "Authorization")
        request.httpMethod = "PUT"
        
        let session = URLSession.shared
        let task = session.dataTask(with: request) // ChatGPT
        try await task.resume() //ChatGPT
        
    }
    
    
    
}
